### ISSUE TITLE
Unexpected error if a $ref cannot be resolved

### ISSUE THREAD

**Issue by k7sleeper** [zaggino/z-schema#11]
Running the following code under Node.js

```
var ZSchema, main, schemaA;
ZSchema = require("z-schema");

schemaA = {
    $schema: 'http://json-schema.org/draft-04/schema#',
    id: "schemaA",
    type: "object",
    properties: {a: {type: "integer"}, b: {type: "string"}, c: {$ref: 'schemaB'}},
    required: ['a', 'c']
  };

var json, schema;
    schema = schemaA;
    json = {a: 4, c: 4};
    console.info("#1: validate ...");
    ZSchema.validate(json, schema, function(err, report) {
      var _ref;
      if (err) {
        console.error("... not valid!");
        if ("errors" in err) {
          console.error(err.errors);
        }
        return;
      }
      console.info("... OK");
      if ((_ref = report.warnings) != null ? _ref.length : void 0) {
        return console.warn(report.warnings);
      }
    });
```

results to the following console output, which I don't expect:

```
#1: validate ...
... not valid!
[ { code: 'INVALID_TYPE',
    message: 'invalid type: integer (expected object)',
    path: '#/c',
    params: { expected: 'object', type: 'integer' } } ]
```
